docs/recipes/probes.md

```diff
@@ -0,0 +1,172 @@
+
+### Readiness  
+
+The `readiness` probe is used to let Tye know when it's okay to route traffic to a replica.  
+
+The `readiness` probe cannot kill/restart a replica, it can only do two things
+
+* Promote `Healthy` replica to `Ready`, when the probe succeeds.
+* Demote a `Ready` replica to `Healthy` when the probe fails.  
+
+Tye only routes traffic (either via a service binding, or an ingress) to `Ready` replicas.  
+
+## Life cycle of a Replica in the Absence of Probes
+
+By now, the life cycle of a replica when both `liveness` and `readiness` probes are configured should be clear, but how does the life cycle of a replica look like when both probes or one of the probes is absent?  
+
+* When both `liveness` and `readiness` probes are absent, a replica gets promoted from `Started` to `Ready` automatically, upon creation.
+* When only the `readiness` probe is absent, a replica gets promoted from `Started` to `Healthy` only after it passes the `liveness` probe, but upon being promoted to `Healthy`, it's automatically promoted again, to `Ready`.  
+* When only the `liveness` probe is absent, a replica gets promoted from `Started` to `Healthy` automatically, upon creation, but gets promoted from `Healthy` to `Ready` only upon passing the `readiness` probe.  
+
+## Running Locally with Liveness and Readiness Probes
+
+*The section will refer to this [sample application](/samples/liveness-and-readiness/) which demonstrates basic usage.*  
+
+To run the sample locally, clone this repository or download the source and navigate to the `/samples/liveness-and-readiness/` directory in your terminal.  
+
+```
+tye run
+```
+
+The sample has a single service with a `liveness` and a `readiness` probe, as shown in this snippet
+
+```
+services:
+  - name: simple-webapi
+    project: webapi/webapi.csproj
+    replicas: 2
+    liveness:
+      http:
+        path: /healthy
+    readiness:
+      http:
+        path: /ready
+```
+
+The service is configured to respond successfully to both the `liveness` and `readiness` probes, so after you run Tye, you should see these log lines  
+
+```
+[18:14:05 INF] Replica simple-webapi_a2d67bd9-4 is moving to an healthy state
+[18:14:05 INF] Replica simple-webapi_a9c2e2f4-d is moving to an healthy state
+[18:14:07 INF] Replica simple-webapi_a9c2e2f4-d is moving to a ready state
+[18:14:07 INF] Replica simple-webapi_a2d67bd9-4 is moving to a ready state
+```
+
+As you can see, both replicas pass the `liveness` probe and get prompted to an `Healthy` state, and shortly after, both replica pass the `readiness` probe and get promoted to a `Ready` state.  
+
+The sample application exposes an endpoint that allows you modify the responses that the `/healthy` and `/ready` endpoints, in order to see the probes in action.  
+
+For example, if you send an *HTTP POST* request to this endpoint `http://localhost:8080/set` with this body  
+
+```
+{
+	"Ready": false,
```

Comment:
Can you use query parameters instead of a post message to populate these values? It's a lot easier for people to create a get request with a url (ex just copying the url into a browser) versus a POST request (using postman or curl).